### PR TITLE
add training label smooth

### PR DESCRIPTION
**Pull request by CPFLAME**
### 这个pr要解决的：
- [x] 在nn_graph中加入label smooth的trick进行训练
- [x] 和pytorch实现的label smooth进行对齐

pytorch的实现源自于[此处](https://github.com/pytorch/pytorch/issues/7455#issuecomment-513062631)
运行python utils/test_labelsmooth.py即可得到结果，误差最大值的量级在e-7, 几乎一样

```
abs diff: 1.1920928955078125e-07, flow labelsmooth loss: 1.3462207317352295, torch labelsmooth loss:1.34622061252594
abs diff: 1.1920928955078125e-07, flow labelsmooth loss: 1.5114572048187256, torch labelsmooth loss:1.511457085609436
abs diff: 0.0, flow labelsmooth loss: 1.6171989440917969, torch labelsmooth loss:1.6171989440917969
abs diff: 0.0, flow labelsmooth loss: 0.6735591292381287, torch labelsmooth loss:0.6735591292381287
abs diff: 0.0, flow labelsmooth loss: 1.9999173879623413, torch labelsmooth loss:1.9999173879623413
abs diff: 1.1920928955078125e-07, flow labelsmooth loss: 1.3488662242889404, torch labelsmooth loss:1.3488661050796509
abs diff: 0.0, flow labelsmooth loss: 1.5050686597824097, torch labelsmooth loss:1.5050686597824097
abs diff: 1.1920928955078125e-07, flow labelsmooth loss: 1.449904203414917, torch labelsmooth loss:1.4499040842056274
abs diff: 0.0, flow labelsmooth loss: 1.8043279647827148, torch labelsmooth loss:1.8043279647827148
abs diff: 5.960464477539063e-08, flow labelsmooth loss: 0.7393254041671753, torch labelsmooth loss:0.7393253445625305
```

### CODE REVIEW

File: resnet50/graph/train_labelsmooth.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+class LabelSmoothLoss(Module):
+    def __init__(self,) -> None:
+        super().__init__()
+    
+    def forward(self, input, label):
+        assert label.size() == input.size()
+        log_prob = input.softmax(dim=-1).log()
+        loss = flow.mul(log_prob * -1, label).sum(dim=-1).mean()
+        return loss
```
Comment (CPFLAME) [inline resnet50/graph/train_labelsmooth.py:55]: one_hot函数中包含有.numpy()的操作，不能放在nn_graph.build()函数中。

File: resnet50/graph/train_labelsmooth.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            # self.train_data_loader = train_data_loader
+        
+        def build(self, image, label):
+            # image, label = self.train_data_loader()
+            image = image.to("cuda")
+            label = label.to("cuda")
+            logits = self.resnet50(image)
+            loss = self.cross_entropy(logits, label)
+            loss.backward()
+            return loss
```
Comment (CPFLAME) [inline resnet50/graph/train_labelsmooth.py:107]: 此处把data_loader从build中取出来，放在nn_graph外部，原因为生成one_hot操作无法在build内部进行

File: resnet50/graph/train_labelsmooth.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    resnet50_module = resnet50()
+    if args.load_checkpoint != "":
+        print("load_checkpoint >>>>>>>>> ", args.load_checkpoint)
+        resnet50_module.load_state_dict(flow.load(args.load_checkpoint))
+
+    end_t = time.time()
+    print("init time : {}".format(end_t - start_t))
+
+    # of_cross_entropy = flow.nn.CrossEntropyLoss()
+    of_cross_entropy = LabelSmoothLoss()
```
Comment (CPFLAME) [inline resnet50/graph/train_labelsmooth.py:83]: 选取loss函数

File: resnet50/graph/train_labelsmooth.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        resnet50_module.train()
+
+        for b in range(len(train_data_loader)):
+            # oneflow graph train
+            image, label = train_data_loader()
+            one_hot_label = one_hot(label, classes=1000, smooth_rate=0.1)
+            start_t = time.time()
+
+            loss = resnet50_graph(image, one_hot_label)
+
```
Comment (CPFLAME) [inline resnet50/graph/train_labelsmooth.py:143]: 此处为训练时的逻辑，需要事先生成好one_hot_label，然后送入nn_graph前向
